Rename Office 365 to Entra ID

marianunez opened the pull request:
#### Summary
 
 Rename Office 365 to Entra ID. No changes in config settings to avoid migrations.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-57435


#### Screenshots

|  before  |  after  |
|----|----|
|<img width="949" alt="Screenshot 2024-07-29 at 12 32 40 PM" src="https://github.com/user-attachments/assets/99dcd8f3-ab2a-42c7-b37a-824fe407b722"> | <img width="1049" alt="Screenshot 2024-07-29 at 4 00 29 PM" src="https://github.com/user-attachments/assets/571148ff-d426-46f1-89cd-dcbb51fa431a"> |

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates).
* API additions—new endpoint, new response fields, or newly accepted request parameters.
* Database changes (any).
* Schema migration changes. Use the [Schema Migration Template](https://docs.google.com/document/d/18lD7N32oyMtYjFrJKwsNv8yn6Fe5QtF-eMm8nn0O8tk/edit?usp=sharing) as a starting point to capture these details as release notes. 
* Websocket additions or changes.
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating).
* New features and improvements, including behavioral changes, UI changes, and CLI changes.
* Bug fixes and fixes of previous known issues.
* Deprecation warnings, breaking changes, or compatibility notes.

If no release notes are required, write NONE. Use past-tense. Newlines are stripped.

Examples:

```
Added new API endpoints POST /api/v4/foo, GET api/v4/foo, and GET api/v4/foo/:foo_id.
```

```
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```
NONE
```
-->
```release-note
 Rename Office 365 to Entra ID for SSO logins
```

marianunez commented:
@esethna For the configuration page of Open ID Office 365, I updated the title but there are some references in the help text as well. I can update those but there is also some links for the configuration that I'm assuming that we don't need to change? I am not familiar with the feature so who can validate these?

<img width="1014" alt="Screenshot 2024-07-29 at 4 05 47 PM" src="https://github.com/user-attachments/assets/fd29b770-dabd-4831-969c-9059c1d0b8c9">

marianunez commented:
@lindalumitchell do you know who (and if it's something we can easily do) to help setup Office 365 to test it and validate some of the name updates for the login and user account settings screens?

lindalumitchell commented:
@marianunez I was able to get this set up through a new flow after poking around a bit. We can currently log in to the PR test server using Office 365 credentials.

Some links in the help text in System Console `/admin_console/authentication/openid` will be outdated. Noting below what worked for me, though I can't be sure varying customer setups would work the same:

1. At least one link in step 1 needs updating
  - `Log in` link to https://login.microsoftonline.com/ still works to get to login screen 
  - But `tenant` link to https://msdn.microsoft.com/en-us/library/azure/jj573650.aspx#Anchor_0 may need to be updated as it refers to Azure AD and shows a banner `We're no longer updating this content regularly`
2. Link in step 2 https://apps.dev.microsoft.com leads to an error page. I used https://entra.microsoft.com instead 
  - Then navigate to Applications and Application Registrations in the left nav
3. Still works: Name the app as described, copy and enter client and directory IDs as described
4. Still works but navigation instructions need updating: Enter the Redirect URI as `your-mattermost-url/signup/office365/complete` (but no longer follow the navigation as described to get there)
5. Still works but navigation instructions need updating: Generating and entering Client secret works as expected, though as above the steps to get there need updating

I'm not sure overall what all needs to reflect new naming / content and what should stay the same, but the above is what I found when I went through the links and the setup process.

esethna commented:
Appreciate the detailed testing Linda. @cwarnermm can you help work with @lindalumitchell to propose updated help text based on the above? Perhaps Schalla can verify once ready as the SME?

cwarnermm commented:
Based on @lindalumitchell's testing, here's updated in-product text draft (where outstanding questions are noted):

1. [Log in](https://login.microsoftonline.com/) to your Microsoft account. Ensure it's the same <tenant> you want users to log in with. 

2. Go to [https://entra.microsoft.com/](https://entra.microsoft.com/), then go to **Applications** and **Application Registrations** on the left.

3. Use ``<Mattermost - your-company-name>`` as the **Application Name**. Select **Registration**, and paste your **Client ID** and **Tenant ID**. 

4. [Where?] Enter ``your-mattermost-url/signup/office365/complete``. For example: ``http://localhost:8065/signup/office365/complete`` Also uncheck **Allow Implicit Flow**. 

5. Select **Certificates & secrets**. Generate a **New client secret**, and paste the secret value in the **Client Secret** field.

Outstanding (SME to confirm):

- Need new <tenant> URL. ``https://msdn.microsoft.com/en-us/library/azure/jj573650.aspx#Anchor_0`` doesn't work.

- Confirm that ``your-mattermost-url/signup/office365/complete`` path should include ``office365`` 

- Confirm UI navigation for all steps, and particularly step 4.

lindalumitchell commented:
Thanks @cwarnermm ! I was a bit imprecise with a couple of my notes, so here are some clarifications on some of the steps (obviously feel free to re-word / reformat as you see fit):

>     2. Go to https://entra.microsoft.com/, then go to **Applications** and **Application Registrations** on the left.
Replace **Application Registrations** with **App Registrations**

>     3. Use `<Mattermost - your-company-name>` as the **Application Name**. Select **Registration**, and paste your **Client ID** and **Tenant ID**.
3. Click **New Registration**, then enter `<Mattermost - your-company-name>` as the **Application Name**.

>     4. [Where?] Enter `your-mattermost-url/signup/office365/complete`. For example: `http://localhost:8065/signup/office365/complete` Also uncheck **Allow Implicit Flow**.
4. Under **Redirect URI**, select **Web** as the platform, and enter `your-mattermost-url/signup/office365/complete` as the **Redirect URI**. Then click **Register**.

>     5. Select **Certificates & secrets**. Generate a **New client secret**, and paste the secret value in the **Client Secret** field.
5. Copy the **Application (client) ID** and **Directory (tenant) ID** and enter them here in the corresponding fields.

6. Next to **Client credentials**, click **Add a certificate or secret**, copy the resulting value, and enter it here in the **Client Secret** field, then **Save**.

amyblais commented:
/update-branch

amyblais commented:
> Thanks @cwarnermm ! I was a bit imprecise with a couple of my notes, so here are some clarifications on some of the steps (obviously feel free to re-word / reformat as you see fit):
> 
> > ```
> > 2. Go to https://entra.microsoft.com/, then go to **Applications** and **Application Registrations** on the left.
> > ```
> 
> Replace **Application Registrations** with **App Registrations**
> 
> > ```
> > 3. Use `<Mattermost - your-company-name>` as the **Application Name**. Select **Registration**, and paste your **Client ID** and **Tenant ID**.
> > ```
> 
> 3. Click **New Registration**, then enter `<Mattermost - your-company-name>` as the **Application Name**.
> 
> > ```
> > 4. [Where?] Enter `your-mattermost-url/signup/office365/complete`. For example: `http://localhost:8065/signup/office365/complete` Also uncheck **Allow Implicit Flow**.
> > ```
> 
> 4. Under **Redirect URI**, select **Web** as the platform, and enter `your-mattermost-url/signup/office365/complete` as the **Redirect URI**. Then click **Register**.
> 
> > ```
> > 5. Select **Certificates & secrets**. Generate a **New client secret**, and paste the secret value in the **Client Secret** field.
> > ```
> 
> 5. Copy the **Application (client) ID** and **Directory (tenant) ID** and enter them here in the corresponding fields.
> 6. Next to **Client credentials**, click **Add a certificate or secret**, copy the resulting value, and enter it here in the **Client Secret** field, then **Save**.

@DSchalla @wiggin77 Would you be open to help review the suggested help text here, and apply any changes in this PR while Maria is on PTO for the next two weeks?

cwarnermm commented:
@DSchalla @wiggin77 - To clarify, please review [this](https://github.com/mattermost/mattermost/pull/27801#issuecomment-2258316343) draft help text.

wiggin77 commented:
> 1. [Log in](https://login.microsoftonline.com/) to your Microsoft account. Ensure it's the same  you want users to log in with.

Ensure _what_ is the same?

wiggin77 commented:
> 4\. Under **Redirect URI**, select **Web** as the platform, and enter `your-mattermost-url/signup/office365/complete` as the **Redirect URI**. Then select **Register**.

Should we provide any guidance around what to set for "Supported account types"?

wiggin77 commented:
> 6\. Next to **Client credentials**, select **Add a certificate or secret**, copy the resulting value, enter it in the **Client Secret** field, then select **Save**.

I think this steps missed an instruction to click "+ New client secret"

amyblais commented:
@cwarnermm Please see Doug's notes above ^

cwarnermm commented:
@wiggin77 

> > 1. [Log in](https://login.microsoftonline.com/) to your Microsoft account. Ensure it's the same  you want users to log in with.
> 
> Ensure _what_ is the same?

Ensure the ``tenant`` is the same: Current ``tenant`` link goes to: ``https://msdn.microsoft.com/en-us/library/azure/jj573650.aspx#Anchor_0`` which doesn't resolve.

wiggin77 commented:
> @wiggin77
> 
> > > 1. [Log in](https://login.microsoftonline.com/) to your Microsoft account. Ensure it's the same  you want users to log in with.
> > 
> > 
> > Ensure _what_ is the same?
> 
> Ensure the `tenant` is the same: Current `tenant` link goes to: `https://msdn.microsoft.com/en-us/library/azure/jj573650.aspx#Anchor_0` which doesn't resolve.

I'm suggesting that is not clear in the docs.

cwarnermm commented:
> > 6. Next to **Client credentials**, select **Add a certificate or secret**, copy the resulting value, enter it in the **Client Secret** field, then select **Save**.
> 
> I think this steps missed an instruction to click "+ New client secret". You then add a description, select a expiry time, and click "Add". It doesn't say what to do with the resulting secret.

@wiggin77 - Would you be open to recording the workflow as you see it? That would be a huge help in developing the in-product help text. Right now, I'm attempting to develop it third-hand, which is challenging.

wiggin77 commented:
> Would you be open to recording the workflow as you see it?

@cwarnermm I don't have time this week to research it as I'm not a subject expert on this.  I just walked through the steps like anyone else would and pointed out what was confusing or not matching what I was seeing.

cwarnermm commented:
@amyblais - Our attempts to have one person test the workflow and another document it isn't working as well as we'd hoped, largely because the workflow is considerably different than the instructions we're working from.

Is there an environment where I can see this workflow for myself? Alternatively, is someone available to review the full end-to-end workflow and provide instructions for the in-product text?

cwarnermm commented:
@esethna - @lindalumitchell also identified an outstanding in-product ``Office 365`` text reference -- the **Authentication Method** shown via **System Console > Users > Specific User** that should change to ``Entra ID``.

![Screenshot 2024-07-31 at 4 25 05 PM](https://github.com/user-attachments/assets/de250917-321c-4d8a-82b1-e14ddf836799)

amyblais commented:
@esethna A kind reminder on this text review https://github.com/mattermost/mattermost/pull/27801#issuecomment-2261432602.

esethna reviewed (approved):
Thanks @cwarnermm and @lindalumitchell!

amyblais commented:
https://github.com/mattermost/mattermost/pull/27801#issuecomment-2261432602

@DSchalla @wiggin77 This is ready to be re-reviewed and any needed code changes applied.

wiggin77 commented:
> Apologizes for the back-and-forth churn on the in-product updates. Thanks to a recording of the workflow by @lindalumitchell, I propose the following in-product help text for the System Console:
> 
> 5. Copy the Microsoft **Application (client) ID** value, and paste it here as the **Client ID** value.
> 6. Copy the Microsoft **Directory (tenant) ID** value, and paste it here as the **Directory (tenant) ID** value.
> 7. In Microsoft, create a new client secret. Copy the resulting client secret value, and paste it here as the **Client Secret** value. Select **Save**.
> 

@cwarnermm  After step 4, we are at a screen like this:
![Screenshot from 2024-08-02 13-51-44](https://github.com/user-attachments/assets/b0ff8058-fcbf-418b-974e-5f338b659495)

I can certainly copy the App ID and tenant ID, but step 5 & 6 say to paste them "here".  Where exactly?  There is nowhere on the screenshot to paste them.

lindalumitchell commented:
> I can certainly copy the App ID and tenant ID, but step 5 & 6 say to paste them "here". Where exactly? There is nowhere on the screenshot to paste them.

@wiggin77 These steps are help text on the /admin_console/authentication/openid page in System Console, so "here" refers to fields on that page.

wiggin77 commented on webapp/channels/src/components/admin_console/admin_definition.tsx:
```diff
@@ -4612,9 +4612,9 @@ const AdminDefinition: AdminDefinitionType = {
+                                    display_name: defineMessage({id: 'admin.oauth.office365', defaultMessage: 'Entra ID'}),
                                     isHidden: it.all(it.not(it.licensedForFeature('Office365OAuth')), it.not(it.cloudLicensed)),
-                                    help_text: defineMessage({id: 'admin.office365.EnableMarkdownDesc', defaultMessage: '1. <linkLogin>Log in</linkLogin> to your Microsoft or Office 365 account. Make sure it`s the account on the same <linkTenant>tenant</linkTenant> that you would like users to log in with.\n2. Go to <linkApps>https://apps.dev.microsoft.com</linkApps>, click <strong>Go to app list ...
+                                    help_text: defineMessage({id: 'admin.office365.EnableMarkdownDesc', defaultMessage: '1. <linkLogin>Log in</linkLogin> to your Microsoft or Entra ID account. Make sure it`s the account on the same <linkTenant>tenant</linkTenant> that you would like users to log in with.\n2. Go to <linkApps>https://entra.microsoft.com</linkApps>, click <strong>Applications > App  ...
```
Typo?  

<strong>Client Secret</strong> fieldhat you would like users to log in  below.'}),

marianunez replied:
Fixed in latest copy

marianunez commented:
@cwarnermm just got back from PTO and catching up on this PR, can you provide the final text changes that are needed after the discussions in this PR?

cwarnermm commented:
@marianunez - Welcome back!! There are 3 changes needed as follows:

1. [Hint text updates](https://github.com/mattermost/mattermost/pull/27801#issuecomment-2261432602) for the System Console page setting.
2. [An outstanding Office 365 text reference](https://github.com/mattermost/mattermost/pull/27801#issuecomment-2261435450)
3. [A hint text typo](https://github.com/mattermost/mattermost/pull/27801#discussion_r1702186023)

marianunez commented:
/update-branch

marianunez commented:
> @marianunez - Welcome back!! There are 3 changes needed as follows:
> 
> 1. [Hint text updates](https://github.com/mattermost/mattermost/pull/27801#issuecomment-2261432602) for the System Console page setting.
> 2. [An outstanding Office 365 text reference](https://github.com/mattermost/mattermost/pull/27801#issuecomment-2261435450)
> 3. [A hint text typo](https://github.com/mattermost/mattermost/pull/27801#discussion_r1702186023)

Thanks @cwarnermm! 

1. Made the help text changes (also fixes the point 3 - typo) and now the update screen is as shown below. Let me know if there are any corrections needed

<img width="936" alt="Screenshot 2024-08-13 at 11 34 10 PM" src="https://github.com/user-attachments/assets/bbd762e9-f737-4f28-b001-99fdf1a0de7b">

2. I've fixed the outstanding text reference in the User Details page in the System Console. Good catch @lindalumitchell, you should be able to validate it in the test sprinwick that is now updated. Let me know if you find anything else.

cwarnermm commented:
Thanks, @marianunez! 

For step #4 of the hint text, please add quotes to: "your-mattermost-url/signup/office365/complete"

There's a small typo in step #6: Directoty should be Directory

marianunez commented:
> For step #4 of the hint text, please add quotes to: "your-mattermost-url/signup/office365/complete"
> 
> There's a small typo in step #6: Directoty should be Directory

@cwarnermm updated. Feel free to take a look at the test server for anything else. 

<img width="1014" alt="Screenshot 2024-08-14 at 1 04 56 PM" src="https://github.com/user-attachments/assets/142e2246-a4a3-452b-942f-2165d314ad26">

marianunez commented:
/update-branch

marianunez commented:
@lindalumitchell @cwarnermm gentle ping on this as we need to include this for v10 release.

lindalumitchell commented on webapp/channels/src/components/user_settings/security/user_settings_security.tsx:
```diff
@@ -495,7 +495,7 @@ export class SecurityTab extends React.PureComponent<Props, State> {
                 <FormattedMessage
                     id='user.settings.security.loginOffice365'
-                    defaultMessage='Login done through Office 365'
+                    defaultMessage='Login done through Entra IDS'
```
```suggestion
                    defaultMessage='Login done through Entra ID'
```

lindalumitchell replied:
Looks like a typo in this line with "IDS" instead of "ID", if I understand the changes correctly?

marianunez replied:
Yes, not sure how that got in there. Good catch! Fixed.

lindalumitchell reviewed (approved):
Awesome, thanks Maria! LGTM.

cwarnermm reviewed (approved):
Thank you, @marianunez!!

mattermost-build commented:
Cherry pick is scheduled.
